cli/src/main/java/com/devonfw/tools/ide/process/ProcessContextImpl.java

```diff
@@ -313,22 +316,34 @@ private String addExecutable(String exec, List<String> args) {
   private void performLogOnError(ProcessResult result, int exitCode, String interpreter) {
 
     if (!result.isSuccessful() && (this.errorHandling != ProcessErrorHandling.NONE)) {
-      String message = createCommandMessage(interpreter, " failed with exit code " + exitCode + "!");
+      IdeSubLogger subLogger = this.context.error();
+      IdeLogLevel ideLogLevel;
+      String message = createCommandMessage(interpreter, "\nfailed with exit code " + exitCode + "!");
+
       if (this.errorHandling == ProcessErrorHandling.THROW_CLI) {
+        subLogger.log(message);
+        result.log(IdeLogLevel.ERROR, context);
         throw new CliProcessException(message, result);
       } else if (this.errorHandling == ProcessErrorHandling.THROW_ERR) {
+        subLogger.log(message);
+        result.log(IdeLogLevel.ERROR, context);
         throw new IllegalStateException(message);
       }
-      IdeSubLogger level;
+
       if (this.errorHandling == ProcessErrorHandling.LOG_ERROR) {
-        level = this.context.error();
+        ideLogLevel = IdeLogLevel.ERROR;
+        subLogger = this.context.error();
       } else if (this.errorHandling == ProcessErrorHandling.LOG_WARNING) {
-        level = this.context.warning();
+        ideLogLevel = IdeLogLevel.WARNING;
+        subLogger = this.context.warning();
       } else {
-        level = this.context.error();
+        ideLogLevel = IdeLogLevel.ERROR;
+        subLogger = this.context.error();
         this.context.error("Internal error: Undefined error handling {}", this.errorHandling);
       }
-      level.log(message);
+
+      subLogger.log(message);
+      result.log(ideLogLevel, context);
```

Comment:
do we probably need two log-levels? One for `out` and one for `err` or should `err` always go on log-level `ERROR`?

Reply:
Removed subLogger and just used result.log now instead.